warehouse/models/mart/transit_database/dim_mobility_mart_providers.sql

```diff
@@ -0,0 +1,160 @@
+    WHERE _is_current
+),
+
+gtfs_services AS (
+    SELECT * FROM {{ ref('dim_gtfs_service_data') }}
+    WHERE _is_current
+),
+
+gtfs_datasets AS (
+    SELECT * FROM {{ ref('dim_gtfs_datasets') }}
+    WHERE _is_current
+),
+
+-- This query will collect every funding source an Organization has via the
+-- services it provides.
+funding_by_organization AS (
+    SELECT
+        orgs.key AS org_key,
+        ARRAY_AGG(services_funding_bridge.funding_program_name) AS funding_sources
+    FROM
+        organizations orgs
+    INNER JOIN
+        orgs_services_bridge ON orgs_services_bridge.organization_key = orgs.key
+    INNER JOIN
+        services_funding_bridge ON orgs_services_bridge.service_key = services_funding_bridge.service_key
+    GROUP BY
+        orgs.key
+),
+
+-- This table has historical data for every year (looks like dating back to
+-- 2018), so let's narrow down that table to *just* the most recent year so
+-- we don't get duplicate rows for each year.
+--
+-- We cannot use `_is_current` here because every year is marked as "current"
+-- since it's the "current" record for the respective year.
+annual_ntd AS (
+    SELECT * FROM {{ ref('dim_annual_ntd_agency_information') }}
+    WHERE state = "CA"
+
+    -- We only want data from the latest data from NTD. In the rare edge case
+    -- that an NTD ID is removed and no longer exists in the current year's
+    -- dataset, then replace `DENSE_RANK()` with `ROW_NUMBER()` and add a
+    -- `PARTITION BY ntd_id` inside of the `OVER()`.
+    QUALIFY DENSE_RANK() OVER(ORDER BY year DESC) = 1
+),
+
+gtfs_data_by_organization AS (
+    SELECT
+        orgs.key AS org_key,
+        ARRAY_AGG(gtfs_datasets.uri) AS gtfs_uris
+    FROM organizations orgs
+    LEFT JOIN
+        orgs_services_bridge ON orgs.key = orgs_services_bridge.organization_key
+    LEFT JOIN
+        gtfs_services ON orgs_services_bridge.service_key = gtfs_services.service_key
+    LEFT JOIN
+        gtfs_datasets ON gtfs_services.gtfs_dataset_key = gtfs_datasets.key
+    WHERE
+        orgs.public_currently_operating IS TRUE
+        AND gtfs_services.customer_facing IS TRUE
+        -- We only want to track schedule feeds for right now
+        AND gtfs_datasets.type = 'schedule'
+    GROUP BY
+        orgs.key
+),
+
+serviced_counties_by_organization AS (
+    SELECT
+        orgs.key AS org_key,
+        ARRAY_CONCAT_AGG(services.operating_counties) AS operating_counties
+    FROM organizations orgs
+    LEFT JOIN
+        orgs_services_bridge ON orgs_services_bridge.organization_key = orgs.key
+    LEFT JOIN
+        services ON services.key = orgs_services_bridge.service_key
+    GROUP BY
+        orgs.key
+),
+
+mobility_market_providers AS (
+    SELECT
+        annual_ntd.agency_name AS agency_name,
+        annual_ntd.ntd_id AS ntd_id,
+        annual_ntd.city AS hq_city,
+        orgs_hq_bridge.county_geography_name AS hq_county,
+        ARRAY(
```

Comment:
Just confirming -- the downstream use case for this is programmatic ingestion of some kind (to make the map)? Asking because we sometimes try to avoid array-type columns in mart tables because Metabase doesn't handle arrays well -- if there's a goal for this table to be useful in Metabase, it may be easier to make a bridge table for the three array columns (counties, funding sources, and schedule URIs) and leave the orgs/NTD mapping in the `dim`. But if there's basically one defined non-Metabase use case it might be ok to leave the arrays.

Reply:
In this case, I used an array because I thought Metabase could handle them. The use case for this data is on the Mobility Marketplace website, which will make this data available as a CSV file and the array values will be flattened + delimited by semi-colons. So I can flatten this data in this table if Metabase isn't going to handle it well, is that how I should proceed?

Reply:
Yeah, that (flattening) might be preferable -- for strings in Metabase you can do a `contains` filter but Metabase can't filter on arrays at all. Here's an [example table in Metabase](https://dashboards.calitp.org/question#eyJkYXRhc2V0X3F1ZXJ5Ijp7ImRhdGFiYXNlIjoyLCJxdWVyeSI6eyJzb3VyY2UtdGFibGUiOjIzODAsImZpZWxkcyI6W1siZmllbGQiLDE4Nzg4LG51bGxdLFsiZmllbGQiLDE4NzkyLG51bGxdLFsiZmllbGQiLDE4Nzg2LG51bGxdXX0sInR5cGUiOiJxdWVyeSJ9LCJkaXNwbGF5IjoidGFibGUiLCJ2aXN1YWxpemF0aW9uX3NldHRpbmdzIjp7fX0=) with a few array columns if you want to test what kinds of functionality you'd want.

Reply:
[Here's a few examples](https://github.com/search?q=repo%3Acal-itp%2Fdata-infra%20array_to_string&type=code) of using the `ARRAY_TO_STRING` function that I think you'd want to use